File: lua unicode escape.lua
-- decode_lua_strings.lua

local function is_printable(s)
  local count = 0
  for i = 1, #s do
    local byte = s:byte(i)
    if byte >= 32 and byte <= 126 or byte == 10 or byte == 13 then
      count = count + 1
    end
  end
  return count / #s >= 0.9
end

local function decode_string(s)
  local f, err = load("return " .. s)
  if not f then return s end
  local ok, result = pcall(f)
  if not ok or type(result) ~= "string" then return s end
  if not is_printable(result) then return s end
  return string.format("%q", result) -- returns it re-escaped and quoted
end

-- Read full input
local input = io.open("input.txt", "r"):read("*a")

-- Replace quoted strings
local output = input:gsub("([\"'])(.-)(%1)", function(q, content, endq)
  local raw = q .. content .. endq
  local decoded = decode_string(raw)
  return decoded or raw
end)

-- Write to output
local out = io.open("output.txt", "w")
out:write(output)
out:close()

print("[+] Decoded strings written to output.txt")

File: lua unicode escape.py
import re
import string

def is_printable(s, threshold=0.9):
    # Check if string is mostly printable
    printable = sum(1 for c in s if c in string.printable)
    return (printable / max(len(s), 1)) >= threshold

def decode_lua_string(s):
    try:
        decoded = bytes(s, "utf-8").decode("unicode_escape")
        if is_printable(decoded):
            return decoded
        else:
            return s  # skip decoding if looks like binary
    except:
        return s  # skip if decoding fails

def decode_strings_in_file(input_file="input.txt", output_file="output.txt"):
    with open(input_file, "r", encoding="utf-8") as f:
        content = f.read()

    def replacer(match):
        original = match.group(0)
        inner = match.group(1)
        decoded_inner = decode_lua_string(inner)
        return f'"{decoded_inner}"'

    pattern = r'"((?:[^"\\]|\\.)*?)"'
    new_content = re.sub(pattern, replacer, content)

    with open(output_file, "w", encoding="utf-8") as f:
        f.write(new_content)

    print("[+] Strings decoded (safe mode). Output saved to:", output_file)

decode_strings_in_file()
